controllers/storagecluster/onboarding_validation_keys_generator_job.go

```diff
@@ -0,0 +1,105 @@
+package storagecluster
+
+import (
+	"fmt"
+	"os"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	"github.com/red-hat-storage/ocs-operator/v4/controllers/util"
+)
+
+const (
+	onboardingValidationKeysGeneratorImage   = "ONBOARDING_VALIDATION_KEYS_GENERATOR_IMAGE"
+	onboardingValidationKeysGeneratorJobName = "onboarding-validation-keys-generator"
+	onboardingValidationPublicKeySecretName  = "onboarding-ticket-key"
+)
+
+type onboardingValidationKeysGeneratorJob struct{}
+
+var _ resourceManager = &onboardingValidationKeysGeneratorJob{}
+
+func (o *onboardingValidationKeysGeneratorJob) ensureCreated(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
+
+	if !instance.Spec.AllowRemoteStorageConsumers {
+		r.Log.Info("Spec.AllowRemoteStorageConsumers is disabled, Job is not created")
+		return reconcile.Result{}, nil
+	}
+	var err error
+	r.Log.Info("Spec.AllowRemoteStorageConsumers is enabled. Creating Onboarding validation key generator job")
+	if os.Getenv(onboardingValidationKeysGeneratorImage) == "" {
+		err = fmt.Errorf("OnboardingSecretGeneratorImage env var is not set")
+		r.Log.Error(err, "No value set for env variable")
+		return reconcile.Result{}, err
+	}
+	actualSecret := &corev1.Secret{}
+	// Creating the job only if public is not found
+	err = r.Client.Get(r.ctx, types.NamespacedName{Name: onboardingValidationPublicKeySecretName,
+		Namespace: instance.Namespace}, actualSecret)
+
+	if errors.IsNotFound(err) {
+		onboardingSecretGeneratorJob := getOnboardingJobObject(instance)
+		err = r.Client.Create(r.ctx, onboardingSecretGeneratorJob)
+		if err != nil {
+			r.Log.Error(err, "failed to create/ensure secret")
```

Comment:
You are not creating the secret here:
```suggestion
			r.Log.Error(err, "failed to create onboarding validation key generator job")
```